### PR TITLE
#167/feature/sf footer

### PR DESCRIPTION
**Pull request by dz3n**
# Related issue
https://github.com/DivanteLtd/storefront-ui/issues/167

# Scope of work
SfFooter refactoring

# Screenshots of visual changes
**Desktop:**
SfFooter_desktop
**Mobile:**
SfFooter_mobile

# Checklist

- [x] I followed [composition rules](https://docs.storefrontui.io/component-rules.html) for my component
- [x] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

File: packages/vue/src/components/organisms/SfFooter/SfFooter.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import Vue from "vue";
+import SfFooterColumn from "./_internal/SfFooterColumn.vue";
+
+Vue.component("SfFooterColumn", SfFooterColumn);
+
+export default {
+  name: "SfFooter",
+  props: {
```
Comment (filrak) [inline packages/vue/src/components/organisms/SfFooter/SfFooter.js]: Shouldn't footer be purely presentational component without any logic? What is the point of this behavior?

File: packages/shared/styles/components/SfList.scss [hunk 2]
```diff
@@ -18,4 +18,16 @@
     font-size: $font-size-big-mobile;
     font-family: $body-font-family-secondary;
   }
+  &__item-padding {
```
Comment (filrak) [inline packages/shared/styles/components/SfList.scss]: It doesn't follow BEM convention. Modifiers are prefixed with `--`. Also padding should be the same for every `SfList` element according to designs. Is it too small for this use case? Check line-height/height proportion with designs please

File: packages/shared/styles/components/SfFooter.scss [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+$footer__title-font-family: $body-font-family-primary !default;
+$footer__column-font-family: $body-font-family-secondary !default;
+$characteristic__description-font-size: $font-size-extra-small-desktop !default;
+$mobile_background-color: #F2F2F2;
+
+.sf-footer {
+  margin-top: $footer__margin-top;
+  margin-left: auto;
+  margin-right: auto;
+  width: 80%;
```
Comment (filrak) [inline packages/shared/styles/components/SfFooter.scss]: why 80?

File: packages/shared/styles/components/SfFooter.scss [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+$footer__margin-top: $spacer-extra-big !default;
+$footer__grid-column-gap: $spacer !default;
+$footer__title-color: $c-dark-primary !default;
+$footer__title-font-family: $body-font-family-primary !default;
+$footer__column-font-family: $body-font-family-secondary !default;
+$characteristic__description-font-size: $font-size-extra-small-desktop !default;
+$mobile_background-color: #F2F2F2;
+
+.sf-footer {
+  margin-top: $footer__margin-top;
```
Comment (filrak) [inline packages/shared/styles/components/SfFooter.scss]: > DO NOT use any outer positioning for components (like outer margins). The way they're positioned in layout should be determined in outer environment.

https://docs.storefrontui.io/contributing/coding-guidelines.html#css-structure

File: packages/shared/styles/components/SfFooter.scss [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  
+  @media (max-width: $mobile-max) {
+    margin-top: unset;
+    margin-left: unset;
+    margin-right: unset;
+    width: unset;
+  }
+}
+
+.sf-footer-row {
```
Comment (filrak) [inline packages/shared/styles/components/SfFooter.scss]: > Use flexbox
https://docs.storefrontui.io/contributing/coding-guidelines.html#css-structure

File: packages/vue/src/components/organisms/SfFooter/_internal/SfFooterColumn.html [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<div class="sf-footer-column" :class="{'sf-footer-column--active': isActive, 'sf-footer-column--not-active': !isActive}">
```
Comment (filrak) [inline packages/vue/src/components/organisms/SfFooter/_internal/SfFooterColumn.html]: just use `SfAccordion`

**Comment by dz3n** [comment]
Suggested changes are done

**Comment by filrak** [comment]
2 minor ones left

File: packages/vue/src/components/organisms/SfFooter/SfFooter.js [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import Vue from "vue";
+import SfFooterColumn from "./_internal/SfFooterColumn.vue";
+
+Vue.component("SfFooterColumn", SfFooterColumn);
+
+export default {
+  name: "SfFooter",
+  data() {
+    return {
+      widthOfWindow: window.innerWidth
```
[the remark this answers is not in the document]
Comment (dz3n) [reply packages/vue/src/components/organisms/SfFooter/SfFooter.js]: This is already removed on previous commit

File: packages/vue/src/components/organisms/SfFooter/SfFooter.html [hunk 8]
```diff
@@ -0,0 +1,10 @@
+<footer>
+    <div class="sf-footer-mobile">
```
Comment (filrak) [inline packages/vue/src/components/organisms/SfFooter/SfFooter.html]: SfFooter should have 1 slot as in the initial description. I think you must've misunderstood my directions, sorry. The idea was to have SfFooterColumn as component taht displayes either Accordion or just a content with title, so the API would look liek this:
```html
<SfFooter>
  <SfFooterColumn title="Title 1">
    content
  </SfFooterColumn>
  <SfFooterColumn title="Title 1">
    content
  </SfFooterColumn>
</SfFooter>

Comment (dz3n) [reply packages/vue/src/components/organisms/SfFooter/SfFooter.html]: Hey @filrak Yes, I didn't completely understand your directions. Right now it is working with this API

Comment (filrak) [reply packages/vue/src/components/organisms/SfFooter/SfFooter.html]: Thank you!

File: packages/shared/styles/components/SfFooter.scss [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+  &--active {
+    background-color: unset;
+  }
+  &__accordion-only {
+    display: none;
+  }
+  &__main-only {
+    display: flex;
+  }
+  &__placeholder {
```
Comment (filrak) [inline packages/shared/styles/components/SfFooter.scss]: what is a purpose of this class?

Comment (filrak) [inline packages/shared/styles/components/SfFooter.scss]: what is a purpose of this class?

File: packages/vue/src/components/organisms/SfFooter/_internal/SfFooterColumn.html [hunk 10]
```diff
@@ -0,0 +1,31 @@
+<div class="sf-footer-column-wrapper">
```
Comment (filrak) [inline packages/vue/src/components/organisms/SfFooter/_internal/SfFooterColumn.html]: why do we need a wrapper?

File: packages/shared/styles/components/SfFooter.scss [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+    display: none;
+  }
+  font-family: $footer__column-content-font-family;
+  &-wrapper {
+    @media screen and (min-width: $tablet-min) {
+      max-width: 24.9999%;
+      flex: 1 1 auto;
+    }
+  }
+  &-desktop {
```
Comment (filrak) [inline packages/shared/styles/components/SfFooter.scss]: we have SCSS classes for this, check `_visibility.scss`

**Comment by dz3n** [comment]
I am using **sf-footer-column__placeholder** because I wanted to have class which can be used for placeholder columns and with this way to keep working with the flex and blank accordion columns, and the user may specify for example that he wants 1 blank column, 1 with content, 1 blank and 1 with content and using this class will not create blank accordion items. And also the class name may improve code readability this way.

SfFooterRow component is added

**Comment by filrak** [comment]
Ok i get it! It makes sense in some situations. Perhaps we can even omit this class and use `:empty` CSS selector ?

**Comment by dz3n** [comment]
Hey @filrak 

The changes are pushed

Removed placeholder class and added flex-end for rows

**Comment by przspa** [comment]
thanks, @dz3n for your support 🚀

**Pull request closed without merging**